### PR TITLE
Rework spackbot: security fixes and use a team for maintainers

### PR DESCRIPTION
**Pull request by tgamblin**
This PR does a number of things:

- [x] Change the bot's log level to info so that we can filter out lower-level logs
- [x] Avoid a security issue: we can't run `spack maintainers` from the PR version of Spack. That was originally designed for GitHub actions, which run untrusted, but now that we're running this in a bot, running untrusted code gives PR submitters a way to run anything with the same privileges as `spackbot`. We should not do that.
- [x] Rework the way maintainers are tracked. Instead of adding them directly as repo collaborators on the installation's repo, add them to a team called `maintainers` there, IF it exists. This makes it easier to track who's been added by the bot, and it lets admins of the repo control what permissions the collaborators get (e.g., they can do `triage`, which is not handled by the GitHub API yet)
- [x] Fix issues with the private key when running this bot as a container -- specifically, fix escaping and quoting on the private key.

### CODE REVIEW

File: spackbot/pr_add_reviewers.py [hunk 1]
```diff
@@ -211,16 +280,6 @@ async def add_reviewers(gh, repository, pull_request, number):
...
 
-        # Ask people to maintain packages that don't have maintainers.
-        if unmaintained_pkgs:
-            # Ask for maintainers
-            # https://docs.github.com/en/rest/reference/issues#create-an-issue-comment
-            comment_body = no_maintainers_comment.format(
-                author=pull_request["user"]["login"],
-                packages_without_maintainers="\n* ".join(sorted(unmaintained_pkgs)),
-            )
-            await gh.post(pull_request["comments_url"], {}, data={"body": comment_body})
```
Comment (adamjstewart) [inline spackbot/pr_add_reviewers.py:222]: Why is this removed?

Comment (tgamblin) [reply spackbot/pr_add_reviewers.py:222]: It’s just moved. See above. It’s actually working: https://github.com/spack/spack/pull/24561

**Review by adamjstewart — APPROVED** [review]
